src/react/reconcilation.js

```diff
@@ -1379,10 +1378,12 @@ export class Reconciler {
       return this._resolveArray(componentType, value, context, branchStatus, evaluatedNode, needsKey);
     } else if (value instanceof ObjectValue && isReactElement(value)) {
       return this._resolveReactElement(componentType, value, context, branchStatus, evaluatedNode, needsKey);
-    } else {
-      let location = optionalStringOfLocation(value.expressionLocation);
-      throw new ExpectedBailOut(`invalid return value from render${location}`);
     }
+    // This value is not a valid return value of a render, but given we might be
+    // in a && condition,  it may never result in a runtime error. Still, if it does
```

Comment:
Nit: just "in a condition" is sufficient. Also double space

Unless this is really specific to `&&` and doesn't happen with `if`?

Reply:
It's specific to `&&` which only occurs with Herman's PR applied. Where we have the the left-side of a `&&` condition to evaluate to something thruthy, that is not a valid React return value. As it evaluates to true, the right-side will always be used.